python/dgl/dataloading/dataloader.py

```diff
@@ -732,10 +710,6 @@ def __init__(self, graph, indices, graph_sampler, device='cpu', use_ddp=False,
             if use_uva:
                 if self.graph.device.type != 'cpu':
                     raise ValueError('Graph must be on CPU if UVA sampling is enabled.')
```

Comment:
Do we need a check that `self.device` is a `cuda` device if `use_uva=True`, now that we can't rely on indices? As it defaults to `cpu` I suspect many users will attempt this combination of inputs. Would it make sense to have something like:
```
if device is None:
  if use_uva:
    device = torch.cuda.current_device()
  else:
    device = self.graph.device
```
Rather than defaulting the device to `cpu`?

Reply:
I think #3822 fixes it.